common/lib/xmodule/xmodule/partitions/tests/test_partitions.py

```diff
@@ -113,13 +115,14 @@ class PartitionTestCase(TestCase):
 
     def setUp(self):
         # Set up two user partition schemes: mock and random
+        self.non_random_scheme = MockUserPartitionScheme(self.TEST_SCHEME_NAME)
+        self.random_scheme = MockUserPartitionScheme("random")
         extensions = [
             Extension(
-                self.TEST_SCHEME_NAME, USER_PARTITION_SCHEME_NAMESPACE,
-                MockUserPartitionScheme(self.TEST_SCHEME_NAME), None
+                self.non_random_scheme.name, USER_PARTITION_SCHEME_NAMESPACE, self.non_random_scheme, None
             ),
             Extension(
```

Comment:
I'm confused by this. MockUserPartitionScheme sets its name in its init method:

https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/partitions/tests/test_partitions.py#L91

I'm not sure why it isn't applying in this case.